### ISSUE TITLE
Convert PR deployment pipeline to be a build -> upload drop -> deploy pipeline

### ISSUE THREAD

**Issue by nguyentvan7** [frzyc/genshin-optimizer#1575]
Currently our forked PRs cannot deploy due to lack of access to secrets in `pull_request` events. We can use `pull_request_target` events, but we cannot reliably get a merge SHA without hacky workarounds (#1327).

As a result, we are not seeing if forked PRs are able to actually build the site using our prod-rules.

Due to this, we should convert our deployment pipeline to just be a build pipeline. It will build the site, then zip the `dist/apps/frontend` folder along with some metadata. Then it will upload that as an artifact (free for public repos).

This step can be done for forked and local branches alike, and would verify the build step works on every PR.

After that, we can have an additional `workflow_run` call that will download the artifact, then deploy that using the metadata contained inside.
`workflow_run` calls from within a `pull_request` target will have access to all repo secrets, and so the deployment will actually be able to succeed for even forked repos. Inside this, we can also have the comment/deployment log maintenance, as that requires secrets.

tl;dr: move to build -> upload then download -> deploy 2-step pipeline. This will allow us to check builds of any pr including forks, and will also allow deployment of forks. It also maintains safety, assuming the person who approves the workflows checks the code before doing so
